### DISCUSSION TITLE
SQL autocompletion based on a database schema

### DISCUSSION THREAD

**Discussion by isNotOkay** [microsoft/monaco-editor#4011] [Q&A]
### Problem Description
We would like to use monaco as SQL-Editor for **SQLite** (no other SQL dialect).

For this we started to implement a new _registerCompletionItemProvider_:

```
monaco.languages.registerCompletionItemProvider('sql', {
    provideCompletionItems: () => ({
      suggestions: [
        {
          label: 'SELECT',
          insertText: 'SELECT',
          kind: monaco.languages.CompletionItemKind.Keyword
        },
		....
	}))
```

This has worked quite well so far, but now we are wondering how to implement autocompletion based on the actual database schema.  

Example from sql client [SQLiteStudio](https://sqlitestudio.pl/):

![sql_autocomplete](https://github.com/microsoft/monaco-editor/assets/781073/d92b4c35-1a6a-443f-a4f4-e27ed068196e)

### Questions
1. Does a completion provider for SQLite already exist, so that we don't have to create our own?
2. How to define autocompletion based on a database schema?

**Comment by 51yu** [comment]
Anyone could shed some light ^^

**Reply by guimabdo** [reply]
Did you find a way? I'm searching for it.
